src/praisonai/tests/unit/recipe/test_recipe_serve_command.py

```
@@ -0,0 +1,83 @@
+    assert serve.call_args.kwargs["config"]["auth"] == "api-key"
+
+
+def test_recipe_serve_rejects_unsupported_auth_mode(monkeypatch):
+    serve = Mock()
+    load_config = Mock(return_value={"auth": "apikey", "api_key": "configured"})
+    serve_module = importlib.import_module("praisonai.recipe.serve")
+    monkeypatch.setattr(serve_module, "serve", serve)
+    monkeypatch.setattr(serve_module, "load_config", load_config)
+
+    with pytest.raises(typer.Exit):
+        _invoke_recipe_serve(host="0.0.0.0", config="serve.yaml")
```

Ruff (0.16.1)

[error] Possible binding to all interfaces

(S104)